src/instructions/gt.rs

```diff
@@ -0,0 +1,328 @@
+use crate::{circuit_io_type::CircuitIOType, ConstraintF};
+use anyhow::{bail, Result};
+use ark_r1cs_std::{
+    prelude::{AllocVar, Boolean},
+    select::CondSelectGadget,
+    R1CSVar,
+};
+use indexmap::IndexMap;
+use simpleworks::marlin::ConstraintSystemRef;
+pub use CircuitIOType::{SimpleBoolean, SimpleUInt16, SimpleUInt32, SimpleUInt64, SimpleUInt8};
+
+pub fn gt(
+    operands: &IndexMap<String, CircuitIOType>,
+    constraint_system: ConstraintSystemRef,
+) -> Result<CircuitIOType> {
+    match operands
+        .values()
+        .collect::<Vec<&CircuitIOType>>()
+        .as_slice()
+    {
+        [SimpleUInt8(left_operand), SimpleUInt8(right_operand)] => {
+            let gt = left_operand.value()? > right_operand.value()?;
+            Ok(SimpleBoolean(Boolean::conditionally_select(
+                &Boolean::new_witness(constraint_system, || Ok(gt))?,
+                &Boolean::<ConstraintF>::TRUE,
+                &Boolean::<ConstraintF>::FALSE,
```

Comment:
I'm unsure if there's a difference between using `Boolean::<ConstraintF>::TRUE` or `Boolean::<ConstraintF>::FALSE` and adding a witness `Boolean`. Maybe @jrchatruc can tell.

Reply:
Yeah, these results should be witnesses

Reply:
After talking with @jrchatruc and @diegokingston we concluded that effectively these `Boolean`s (`true_value` and `false_value` in `conditionally_select`) should be witnesses because they may be intermediate variables that could be used to affect the constraint system.